pytorch_lightning/callbacks/stochastic_weight_avg.py

```diff
@@ -145,6 +152,9 @@ def on_fit_start(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule"):
         optimizers = trainer.optimizers
         lr_schedulers = trainer.lr_schedulers
 
+        if isinstance(trainer.training_type_plugin, (DDPFullyShardedPlugin, DeepSpeedPlugin)):
+            raise MisconfigurationException("SWA does not currently support sharded models.")
```

Comment:
@ananthsub @daniellepintz Here is an example where we would have implemented `on_init_end` to fail early instead of spawning processes and then error on fit start.

Reply:
would using `setup` work?